agenta-web/src/components/TestSetTable/ConfirmTestsetDeleteModal.tsx

```diff
@@ -0,0 +1,76 @@
+import React, {useState} from "react"
+import {Modal} from "antd"
+import {ExclamationCircleOutlined} from "@ant-design/icons"
+import {createUseStyles} from "react-jss"
+import {deleteTestsets, convertTestsetsToDummyIfInUse} from "@/lib/services/api"
+
+const useStyles = createUseStyles({
+    modalContainer: {
+        display: "flex",
+        flexDirection: "row",
+        alignItems: "center",
+    },
+    modalIcon: {
+        fontSize: "24px",
+        color: "#faad14",
+        marginRight: "10px",
+    },
+})
+
+interface Props {
+    isModalOpen: boolean
+    setIsModalOpen: (value: boolean) => void
+    testsetsIds: Array<string>
+    mutate: () => void
+    setSelectedRowKeys: (keys: React.Key[]) => void
+}
+
+const ConfirmTestsetDeleteModal: React.FC<Props> = ({
+    isModalOpen,
+    setIsModalOpen,
+    testsetsIds,
+    mutate,
+    setSelectedRowKeys,
+}) => {
+    const classes = useStyles()
+    const [isLoading, setIsLoading] = useState(false)
+    const handleCloseModal = () => setIsModalOpen(false)
+    const handleDismiss = () => handleCloseModal()
+    const handleConfirmOK = async () => {
+        setIsLoading(true)
+        try {
+            // make sure promise is completely fulfilled before deleting testsets
+            await convertTestsetsToDummyIfInUse(testsetsIds).then(async () => {
```

Comment:
Shouldn't this happen automatically on the backend whenever tests are requested to be deleted? I believe this logic should be encapsulated on the backend within the `DELETE /testsets` endpoint

@aakrem thoughts?

Reply:
Thank you for bringing this up! I'll make the necessary changes.